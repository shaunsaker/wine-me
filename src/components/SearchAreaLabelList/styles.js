import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  contentContainer: {},
  labelContainer: {
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
});

export default styles;
