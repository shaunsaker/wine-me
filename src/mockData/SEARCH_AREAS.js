import SEARCH_AREA from './SEARCH_AREA';

const SEARCH_AREAS = { 1: SEARCH_AREA };

export default SEARCH_AREAS;
