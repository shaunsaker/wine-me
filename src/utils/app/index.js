import createError from './createError';
import createMapsLinkFromCoordinates from './createMapsLinkFromCoordinates';
import getDistanceBetweenCoordinates from './getDistanceBetweenCoordinates';
import getGooglePlacesPhotoURI from './getGooglePlacesPhotoURI';
import log from './log';
import navigate from './navigate';
import prepareNextAction from './prepareNextAction';

// All app-specific utils go here
const app = {
  createError,
  createMapsLinkFromCoordinates,
  getDistanceBetweenCoordinates,
  getGooglePlacesPhotoURI,
  log,
  navigate,
  prepareNextAction,
};

export default app;
