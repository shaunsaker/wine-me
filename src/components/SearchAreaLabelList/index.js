import React from 'react';
import PropTypes from 'prop-types';
import { View, FlatList } from 'react-native';

import styles from './styles';

import Label from '../Label';

export default class SearchAreaLabelLIst extends React.Component {
  constructor(props) {
    super(props);

    this.renderItem = this.renderItem.bind(this);
  }

  static propTypes = {
    data: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
      }),
    ),
    handlePress: PropTypes.func,
    testIDPrefix: PropTypes.string,
  };

  static defaultProps = {};

  renderItem({ item }) {
    const { handlePress, testIDPrefix } = this.props;

    return (
      <View style={styles.labelContainer}>
        <Label
          text={item.name}
          handlePress={() => handlePress(item)}
          testID={testIDPrefix + item.name}
        />
      </View>
    );
  }

  render() {
    const { data } = this.props;

    return (
      <FlatList
        keyExtractor={({ id }) => id}
        data={data}
        renderItem={this.renderItem}
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
        horizontal
      />
    );
  }
}
