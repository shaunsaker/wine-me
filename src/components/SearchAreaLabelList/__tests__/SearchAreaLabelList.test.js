import React from 'react';
import renderer from 'react-test-renderer';

import SearchAreaLabelList from '..';
import SEARCH_AREAS from '../../../mockData/SEARCH_AREAS';
import utils from '../../../utils';

describe('SearchAreaLabelList', () => {
  const searchAreasArray = utils.objects.convertObjectToArray(SEARCH_AREAS);

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<SearchAreaLabelList data={searchAreasArray} />);

      expect(component).toMatchSnapshot();
    });

    // NOTE: We don't test render methods (those are handle in the test(s) above)
  });
});
