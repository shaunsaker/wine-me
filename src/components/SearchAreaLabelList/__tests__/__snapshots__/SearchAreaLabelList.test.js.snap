// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAreaLabelList renders renders with minimum required props 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  data={
    Array [
      Object {
        "id": "1",
        "location": Object {
          "lat": -33.9263749,
          "lng": 18.9319848,
        },
        "name": "Robertson",
        "radius": 10,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "marginRight": 7.5,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#E23E57",
              "borderColor": null,
              "borderRadius": 5,
              "borderWidth": 0,
              "elevation": 2,
              "opacity": 1,
              "paddingHorizontal": 7.5,
              "paddingVertical": 5,
              "shadowColor": "#212121",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.17,
              "shadowRadius": 2,
            }
          }
          testID="undefinedRobertson"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": undefined,
                "fontSize": 11,
                "fontWeight": "600",
              }
            }
          >
            Robertson
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
